src/Analyzers/VisualBasic/Tests/RemoveUnnecessaryCast/RemoveUnnecessaryCastTests.vb

```diff
@@ -934,6 +934,23 @@ End Module
             Await TestMissingAsync(markup)
         End Function
 
+        <Fact>
+        Public Async Function TestDoNotRemoveNecessaryCastPassedToIllegalParamArray() As Task
+            Dim markup =
+<File>
+Module M
+    Sub Main()
+        Goo([|CObj(Nothing)|])
+    End Sub
+    Sub Goo(ParamArray x As Object)
```

Comment:
illegal.  but code cannot assume that a paramarray param will def be an arraytypesymbol.